src/main/java/org/ow2/proactive/catalog/report/CatalogObjectReportPDFGenerator.java

```diff
@@ -80,62 +100,150 @@
         }
     }
 
-    private BaseTable createAndIntitializeBaseTable(PDDocument doc, PDPage page) throws IOException {
-        //Initialize table
-        float margin = 10;
+    private void populateTable(List<String[]> catalogObjects, BaseTable table)
+            throws MalformedURLException, IOException {
+        for (String[] catalogObjectData : catalogObjects) {
+
+            Row<PDPage> dataRow = table.createRow(10f);
+            createDataCell(dataRow, (100 / 8f), catalogObjectData[0]);
+            createDataCell(dataRow, (100 / 8f), catalogObjectData[1]);
+            createDataCell(dataRow, (100 / 8f), catalogObjectData[2]);
+            createDataCell(dataRow, (100 / 8f) * 2, catalogObjectData[3]);
+            createDataCell(dataRow, (100 / 8f), catalogObjectData[4]);
+            createDataCell(dataRow, (100 / 8f), catalogObjectData[5]);
+
+            createIconCell(dataRow, catalogObjectData[6]);
+
+        }
+    }
+
+    private BaseTable initializeTable(PDDocument doc, float margin, PDPage page) throws IOException {
         float tableWidth = page.getMediaBox().getWidth() - (2 * margin);
         float yStartNewPage = page.getMediaBox().getHeight() - (2 * margin);
+        boolean drawContent = true;
+        boolean drawLines = true;
         float yStart = yStartNewPage;
-        float bottomMargin = 0;
+        float bottomMargin = 70;
+        BaseTable table = new BaseTable(yStart,
+                                        yStartNewPage,
+                                        bottomMargin,
+                                        tableWidth,
+                                        margin,
+                                        doc,
+                                        page,
+                                        drawLines,
+                                        drawContent);
+        return table;
+    }
 
-        return new BaseTable(yStart, yStartNewPage, bottomMargin, tableWidth, margin, doc, page, true, true);
+    private void createIconCell(Row<PDPage> dataRow, String url_path) throws MalformedURLException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.MalformedURLException' which is a subclass of 'java.io.IOException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)